### ISSUE TITLE
ComplexArticleQuery ignores arguments

### ISSUE THREAD

**Issue by joelhulen** [EventRegistry/event-registry-node-js#1]
When I pass in arguments to the `ComplexArticleQuery` object, they appear to be ignored. I am trying to skip duplicate articles.

The `ComplexArticleQuery` class in query.ts appears to only set the `$filter` value if it is empty:

[/src/query.ts#L198](https://github.com/EventRegistry/event-registry-node-js/blob/64dc6793714699b2221f346bdaf68f16ed10861b/src/query.ts#L198)

``` javascript
if (_.isEmpty(filter)) {
    this.queryObj["$filter"] = filter;
}
```

I believe this should probably be changed to:

``` javascript
if (!_.isEmpty(filter)) {
    this.queryObj["$filter"] = filter;
}
```

This is how I'm currently passing in my arguments from my application:

``` javascript
    const cq = new erBase.ComplexArticleQuery(combinedQueryBase, {
      isDuplicateFilter: 'skipDuplicates',
      hasDuplicateFilter: 'keepAll',
      eventFilter: 'keepAll'
    });
    const q = erBase.QueryArticles.initWithComplexQuery(cq);
```

Despite setting `isDuplicateFilter: 'skipDuplicates'`, I still receive duplicate articles.

Can you please validate whether I am doing something wrong, or if this is a bug as I have outlined?

**Comment by UnspokenMallard** [comment]
@joelhulen You are right, it is a bug. This issue is fixed with the latest version (1.0.91).

**Comment by joelhulen** [comment]
Thank you for the quick resolution! It appears to have worked :)
